### PR TITLE
Modified Product model according to v19 API

### PR DESCRIPTION
**Pull request by marcserrat**

### CODE REVIEW

File: src/Product/Capabilities/PCBilling.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * This file is part of the Ingram Micro Cloud Blue Connect SDK.
+ *
+ * @copyright (c) 2020. Ingram Micro. All Rights Reserved.
+ */
+
+namespace Connect\Product\Capabilities;
+
+use Connect\Model;
+
+class PCBilling extends Model
```
Comment (vgrebenschikov) [inline src/Product/Capabilities/PCBilling.php]: PC prefix is really required here?

File: src/Account.php [hunk 2]
```diff
@@ -3,7 +3,7 @@
 /**
  * This file is part of the Ingram Micro Cloud Blue Connect SDK.
  *
- * @copyright (c) 2019. Ingram Micro. All Rights Reserved.
```
Comment (vgrebenschikov) [inline src/Account.php]: why not 2019 - 2020
(there and in other places)

File: src/Model.php [hunk 3]
```diff
@@ -170,7 +170,7 @@ public static function modelize($key, $value)
         switch (gettype($value)) {
             case 'object':
 
-                $namespaces = ['\Connect\\', '\Connect\Usage\\'];
+                $namespaces = ['\Connect\\', '\Connect\Usage\\', '\Connect\Product\Capabilities\\', 'Connect\Product\Actions\\', 'Connect\Subscription\\'];
```
Comment (vgrebenschikov) [inline src/Model.php]: line too long? probably worth to put every namespace on own line

File: src/Modules/Directory.php [hunk 4]
```diff
@@ -103,7 +105,7 @@ public function listProducts($filters = null)
 
     public function getProduct($productID)
     {
-        $body = $this->sendRequest('GET', '/products/'.$productID);
+        $body = $this->sendRequest('GET', '/products/' . $productID);
```
Comment (vgrebenschikov) [inline src/Modules/Directory.php]: why not via constant like in tier-configs?

File: src/Modules/Directory.php [hunk 5]
```diff
@@ -142,8 +144,94 @@ public function listTierConfigs($filters = null)
...
+     */
+    public function listTierAccounts($filters = null)
+    {
+        if ($filters instanceof \Connect\RQL\Query) {
+            $query = $filters;
+        } elseif (is_array($filters)) {
+            $query = new \Connect\RQL\Query($filters);
+        } else {
+            $query = new \Connect\RQL\Query();
+        }
```
Comment (vgrebenschikov) [inline src/Modules/Directory.php]: probably worth to simplify it as 
``
$query = new \Connect\RQL\Query($filters) 
``
and hide all logic in Query constructor, I seen exactly such block in other places

File: src/Modules/Directory.php [hunk 6]
```diff
@@ -142,8 +144,94 @@ public function listTierConfigs($filters = null)
...
+     */
+    public function listTierAccountRequests($filters = null)
+    {
+        if ($filters instanceof \Connect\RQL\Query) {
+            $query = $filters;
+        } elseif (is_array($filters)) {
+            $query = new \Connect\RQL\Query($filters);
+        } else {
+            $query = new \Connect\RQL\Query();
+        }
```
Comment (vgrebenschikov) [inline src/Modules/Directory.php]: see above about query

File: src/Modules/Subscriptions.php [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+    public function listSubscriptionAssets($filters = null)
+    {
+        $query = new \Connect\RQL\Query();
+
+        if (is_array($filters)) {
+            $query = new \Connect\RQL\Query($filters);
+        } elseif ($filters instanceof \Connect\RQL\Query) {
+            $query = $filters;
+        }
```
Comment (vgrebenschikov) [inline src/Modules/Subscriptions.php]: see above about Query

File: src/Modules/Subscriptions.php [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    public function listSubscriptionRequests($filters = null)
+    {
+        $query = new \Connect\RQL\Query();
+
+        if (is_array($filters)) {
+            $query = new \Connect\RQL\Query($filters);
+        } elseif ($filters instanceof \Connect\RQL\Query) {
+            $query = $filters;
+        }
+
```
Comment (vgrebenschikov) [inline src/Modules/Subscriptions.php]: Query again

File: src/Product.php [hunk 9]
```diff
@@ -117,7 +166,90 @@ public function getProductConfigurations(RQL\Query $filter = null)
...
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getAllItems(RQL\Query $filter = null)
+    {
+        if ($this->id == null) {
+            return [];
+        }
+        if (!$filter) {
+            $filter = new \Connect\RQL\Query();
+        }
```
Comment (vgrebenschikov) [inline src/Product.php]: Query again

File: src/Product/Capabilities/PCBilling.php (hunk 1 above)
Comment (marcserrat) [reply src/Product/Capabilities/PCBilling.php]: Yes, 100% due collisions on how Connect models same models in different places

File: src/Account.php (hunk 2 above)
Comment (marcserrat) [reply src/Account.php]: All files was updated as part of this PR......
no 2019 nowhere

File: src/Model.php (hunk 3 above)
Comment (marcserrat) [reply src/Model.php]: OK

File: src/Modules/Directory.php (hunk 4 above)
Comment (marcserrat) [reply src/Modules/Directory.php]: Sure.....this had been introduced and i'm adopting everywhere when possible....
i added in Directory usage of constants now.

**Pull request merged**